Framework/Core/include/Framework/HistogramRegistry.h

```diff
@@ -161,6 +161,34 @@ std::unique_ptr<TH3> createTH3FromSpec(HistogramSpec const& spec)
   return std::make_unique<T>(spec.name.data(), spec.readableName.data(), spec.config.axes[0].nBins, spec.config.axes[0].bins.data(), spec.config.axes[1].nBins, spec.config.axes[1].bins.data(), spec.config.axes[2].nBins, spec.config.axes[2].bins.data());
 }
 
+/// Helper functions to fill histograms with expressions
+template <typename C1, typename C2, typename C3, typename T>
+void fill(TH1* hist, const T& table, const o2::framework::expressions::Filter& filter)
+{
+  auto filtered = o2::soa::Filtered<T>{{table.asArrowTable()}, o2::framework::expressions::createSelection(table.asArrowTable(), filter)};
+  for (auto& t : filtered) {
+    hist->Fill(*(static_cast<C1>(t).getIterator()), *(static_cast<C2>(t).getIterator()), *(static_cast<C3>(t).getIterator()));
+  }
+}
+
+template <typename C1, typename C2, typename T>
+void fill(TH1* hist, const T& table, const o2::framework::expressions::Filter& filter)
+{
+  auto filtered = o2::soa::Filtered<T>{{table.asArrowTable()}, o2::framework::expressions::createSelection(table.asArrowTable(), filter)};
+  for (auto& t : filtered) {
+    hist->Fill(*(static_cast<C1>(t).getIterator()), *(static_cast<C2>(t).getIterator()));
+  }
+}
+
+template <typename C, typename T>
+void fill(TH1* hist, const T& table, const o2::framework::expressions::Filter& filter)
+{
+  auto filtered = o2::soa::Filtered<T>{{table.asArrowTable()}, o2::framework::expressions::createSelection(table.asArrowTable(), filter)};
+  for (auto& t : filtered) {
+    hist->Fill(*(static_cast<C>(t).getIterator()));
```

Comment:
It may be more efficient to use `FillN` method of ROOT histogram, that allows to fill it from an array

Reply:
Ok, do I see well there is no FillN() for TH3? And one needs to give weights array as well?

Reply:
Not sure how it is implemented for dimensions > 1, I'll have to check. For 1-dimensional you can pass `nullptr` for the weights.